src/test/javascript/spec/service/submission.service.spec.ts

```diff
@@ -94,6 +102,92 @@ describe('Submission Service', () => {
         tick();
     }));
 
+    it('should handle feedback correction round tag', () => {
+        const firstFeedback: Feedback = {
+            id: 3,
+            detailText: 'Feedback',
+            credits: 4,
+            type: FeedbackType.MANUAL,
+        };
+
+        const secondFeedback: Feedback = {
+            id: 4,
+            detailText: 'Feedback',
+            credits: 4,
+            type: FeedbackType.MANUAL,
+        };
+
+        const firstResult: Result = {
+            id: 3556,
+            resultString: '3 of 12 points',
+            score: 24,
+            rated: true,
+            hasFeedback: true,
+            hasComplaint: false,
+            feedbacks: [firstFeedback],
+        };
+
+        submission.results?.unshift(firstResult);
+
+        expect(secondFeedback.copiedFeedbackId).toBe(undefined);
+
+        const latestResultFeedbacks = getLatestSubmissionResult(submission)!.feedbacks!;
+        latestResultFeedbacks?.push(secondFeedback);
+
+        service.handleFeedbackCorrectionRoundTag(0, submission);
+        expect(secondFeedback.copiedFeedbackId).toBe(undefined);
+
+        service.handleFeedbackCorrectionRoundTag(1, submission);
```

Comment:
```suggestion
        // only the second feedback has identical values to the first one, the other feedback should remain untouched
        service.handleFeedbackCorrectionRoundTag(1, submission);
```
Without looking at the service code it was hard to understand why you sometimes assert `undefined` and sometimes a value for the `copiedId`. Some small comment with a short explanation near one of the assertions would help to understand it without having to look at the implementation.

Reply:
Yes I agree, added comments to explain the purpose of each call to `handleFeedbackCorrectionRoundTag`.